n3fit/src/n3fit/model_gen.py

```diff
@@ -782,6 +786,31 @@ def layer_generator(i_layer, nodes_out, activation):
                 )
             return layers
 
+    elif architecture == "tensor_network":
```

Comment:
```suggestion
    elif layer_type == "tensor_network":
```
this should be a ``layer_type`` right?